### DISCUSSION TITLE
Rich text editor: add support for plugins

### DISCUSSION THREAD

**Discussion by cord** [livewire/flux#739] [Feature Requests] [answered]
Similar to the @... mention function on github this utility allows to insert variables/placeholders:

https://tiptap.dev/docs/editor/api/utilities/suggestion

would be great to have that integrated!

**Comment by mauritskorse** [comment]
It would be great if the editor js could be extended to allow using other (even custom) plugins as well.

**Comment by philippbuerger** [comment]
Text align justify is a big thing for many of my customers.

**Comment by philippbuerger** [comment]
A link target `_blank` option would be also nice.

**Comment by joshhanley** [comment, marked as answer]
We have added support for custom extensions in Flux v2.1.3 and added some more hooks in Flux v2.1.6.

Adding custom extensions has been documented here https://fluxui.dev/components/editor#extensions

The new hooks haven't been documented yet, but you can find details here https://github.com/livewire/flux/issues/1490#issuecomment-2845772587
